charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -226,3 +226,19 @@ Get ConfigMap name if existingName is defined, otherwise use default name for ge
   {{- end -}}
 {{- end }}
 
+{{/*
+Create ConfigMap checksum annotation if configMap.configTemplate is defined, otherwise use default file for generated config.
+*/}}
+{{- define "opentelemetry-collector.configTemplateChecksumAnnotation" -}}
+  {{- if .Values.configMap.existingName -}}
+  checksum/config: {{ include (print $.Template.BasePath "/" .Values.configMap.existingName ".yaml" ) . | sha256sum }}
```

Comment:
If a user has `.Values.configMap.existingName = my-favorite-config.yaml` will an extra `.yaml` be added?

Reply:
Yes. I just realized it will work better to just `include` the external template and pass to `cksum` here.